components/server/src/user/user-controller.ts

```diff
@@ -281,6 +286,95 @@ export class UserController {
                 res.sendStatus(200);
             },
         );
+
+        router.post(
+            "/auth/sendHeartbeat/:instanceID",
```

Comment:
not sure why `auth`? 

We should call it not `sendHeartbeat` but something `logWorkspacePageClose`. It is not about heartbeating, but recording a fact that the page was closed.